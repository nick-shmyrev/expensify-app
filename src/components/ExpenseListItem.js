import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <li>
    {new Date(createdAt).toDateString()}: <Link to={`/edit/${id}`} title="Edit Expense"><b>{description}</b></Link>, ${amount / 100}
  </li>
);

export default ExpenseListItem;
