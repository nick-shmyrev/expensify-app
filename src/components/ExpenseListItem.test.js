import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from './ExpenseListItem';

describe('ExpenseListItem component', () => {
  const expense = {
    id: '1234-abcd',
    description: 'Expense to remove',
    note: 'With a note',
    amount: 13,
    createdAt: 0,
  };
  
  test('should render correctly', () => {
    const wrapper = shallow(<ExpenseListItem {...expense}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
