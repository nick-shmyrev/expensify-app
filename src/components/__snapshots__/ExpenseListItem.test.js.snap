// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseListItem component should render correctly 1`] = `
<li>
  Wed Dec 31 1969
  : 
  <Link
    replace={false}
    title="Edit Expense"
    to="/edit/1234-abcd"
  >
    <b>
      Expense to remove
    </b>
  </Link>
  , $
  0.13
</li>
`;
